### PR TITLE
[OptionGroup] Add exception on requests that specify OptionGroupName

### PR DESCRIPTION
**Pull request by Outige**
This PR adds an exception to OptionGroup. When an OptionGroupName is specified.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-rds-optiongroup/src/test/java/software/amazon/rds/optiongroup/CreateHandlerTest.java [hunk 1]
```diff
@@ -112,41 +119,31 @@ public void handleRequest_CreateSuccess() {
...
-        assertThat(response.getErrorCode()).isNull();
+        assertThat(response.getMessage()).isEqualTo("Encountered unsupported property OptionGroupName");
+        assertThat(response.getErrorCode()).isEqualTo(HandlerErrorCode.InvalidRequest);
 
-        verify(proxyClient.client(), times(1)).createOptionGroup(any(CreateOptionGroupRequest.class));
-        verify(proxyClient.client(), times(1)).describeOptionGroups(any(DescribeOptionGroupsRequest.class));
-        verify(proxyClient.client(), times(1)).listTagsForResource(any(ListTagsForResourceRequest.class));
+        verify(proxyClient.client(), times(0)).createOptionGroup(any(CreateOptionGroupRequest.class));
+        verify(proxyClient.client(), times(0)).describeOptionGroups(any(DescribeOptionGroupsRequest.class));
+        verify(proxyClient.client(), times(0)).listTagsForResource(any(ListTagsForResourceRequest.class));
```
Comment (osdrv) [inline aws-rds-optiongroup/src/test/java/software/amazon/rds/optiongroup/CreateHandlerTest.java:143]: Nit: shall it be: `verifyNoMoreInteractions(rdsClient);`

**Review by osdrv — DISMISSED** [review]

File: aws-rds-optiongroup/src/test/java/software/amazon/rds/optiongroup/CreateHandlerTest.java [hunk 2]
```diff
@@ -72,10 +74,15 @@ public void setup() {
         proxy = new AmazonWebServicesClientProxy(logger, MOCK_CREDENTIALS, () -> Duration.ofSeconds(600).toMillis());
         rdsClient = mock(RdsClient.class);
         proxyClient = MOCK_PROXY(proxy, rdsClient);
+        setupResourceModelsForTests();
     }
 
     @AfterEach
-    public void tear_down() {
+    public void tear_down(TestInfo testInfo) {
+        if (testInfo.getTags().contains("IgnoreVerification") == true) {
```
Comment (khebul) [inline aws-rds-optiongroup/src/test/java/software/amazon/rds/optiongroup/CreateHandlerTest.java:82]: nit: Can you make it into constant?

**Review by khebul — APPROVED** [review]

File: aws-rds-optiongroup/src/test/java/software/amazon/rds/optiongroup/AbstractTestBase.java [hunk 3]
```diff
@@ -181,4 +173,8 @@ protected ProgressEvent<ResourceModel, CallbackContext> invokeHandleRequest(
     ) {
         return getHandler().handleRequest(getProxy(), request, context, getProxyClient(), logger);
     }
+
+    protected void setupResourceModelsForTests() {
```
Comment (khebul) [inline aws-rds-optiongroup/src/test/java/software/amazon/rds/optiongroup/AbstractTestBase.java:177]: Question: why do we need this function? My assumption is that by default option group name is null in the model. Is it incorrect?

Comment (Outige) [reply aws-rds-optiongroup/src/test/java/software/amazon/rds/optiongroup/AbstractTestBase.java:177]: My observation is that the default value is null. But it gets overridden by the random OptionGroupName generated by each individual test. Then subsequent calls using RESOURCE_MODEL fail due to there being an OptionGroupName present.

Please let me know if you have a better solution to get around this issue or if you have any follow up questions.

Comment (khebul) [reply aws-rds-optiongroup/src/test/java/software/amazon/rds/optiongroup/AbstractTestBase.java:177]: I have a slight concern that OptionGroup unit tests have side effects on Resource models shared by tests. This means that we are testing resource models that might differ from what we expect.

One option to fix this is to use a getter function that will create a new object every time it is invoked.

**Review by khebul — APPROVED** [review]

**Review by moataz-mhmd — APPROVED** [review]

**Pull request merged**